### ISSUE TITLE
kline.CalcDateRanges panic

### ISSUE THREAD

**Issue by angelinazf** [thrasher-corp/gocryptotrader#545]
https://github.com/thrasher-corp/gocryptotrader/blob/870c8cb90e3c67840bf83456b5bbd8edb5a6665d/exchanges/kline/kline.go#L284-L289

I think here should be 

```
if allDateIntervals != nil && lastNum+1 < len(allDateIntervals) {
	out = append(out, DateRange{
		Start: allDateIntervals[lastNum+1],
		End:   allDateIntervals[len(allDateIntervals)-1],
	})
}
```

because sometimes lastNum is the last index of allDateIntervals, so it will case index out of range panic here.

**Comment by xtda** [comment]
Thanks good pick up! I have added a fix into #522 that is under review at the moment

**Comment by thrasher-** [comment]
#522 has been merged in which includes this fix
